aviary/utils/propeller_map_conversion.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/python
+
+import argparse
+import getpass
+import numpy as np
+from datetime import datetime
+from enum import Enum
+
+from aviary.utils.conversion_utils import conv_rep, conv_parse, read_map
+from aviary.api import NamedValues
+from aviary.utils.csv_data_file import write_data_file
+from aviary.utils.functions import get_path
+from aviary.subsystems.propulsion.utils import default_units
+
+
+class PropMapType(Enum):
+    GASP = 'GASP'
+
+    def __str__(self):
+        return self.value
+
+
+class PropModelVariables(Enum):
+    '''
+    Define constants that map to supported variable names in an propeller model.
+    '''
+    MACH = 'Mach_Number'
+    CP = 'CP'  # power coefficient
+    CT = 'CT'  # thrust coefficient
+    J = 'J'  # advanced ratio
+
+    @classmethod
+    def keys(cls):
+        return [c for c in cls]
+
+
+default_units = {
+    PropModelVariables.MACH: 'unitless',
+    PropModelVariables.CP: 'unitless',
+    PropModelVariables.CT: 'unitless',
+    PropModelVariables.J: 'unitless',
+}
+
+MACH = PropModelVariables.MACH
+CP = PropModelVariables.CP
+CT = PropModelVariables.CT
+J = PropModelVariables.J
+
+# gasp_keys = [MACH, CP, CT, J]
+
+header_names = {}
+for k in PropModelVariables.keys():
+    header_names[k] = k.value
+
+
+def PropDataConverter(input_file, output_file, data_format: PropMapType):
+    """This is a utility class to convert a propeller map file to Aviary format.
+    Currently, there is only one option: from GASP format to Aviary format.
+    As an Aviary command, the usage is:
+    aviary convert_prop_table -F GASP input_file output_file
+    """
+    timestamp = datetime.now().strftime('%m/%d/%y at %H:%M')
+    user = getpass.getuser()
+    comments = []
+    header = {}
+    data = {}
+
+    data_file = get_path(input_file)
+
+    comments.append(f'# created {timestamp} by {user}')
+    comments.append(f'# {data_format} propeller map converted from {input_file}')
+
+    if data_format is PropMapType.GASP:
+        scalars, tables, fields = _read_gasp_propeller(data_file, comments)
+
+        data[J] = tables['thrust_coefficient'][:, 2]
+        data[MACH] = tables['thrust_coefficient'][:, 0]
+        data[CP] = tables['thrust_coefficient'][:, 1]
+        data[CT] = tables['thrust_coefficient'][:, 3]
+
+        # define header now that we know what is in the propeller map
+        # header = {key: default_units[key] for key in gasp_keys}
+
+        # data needs to be string so column length can be easily found later
+        for var in data:
+            data[var] = np.array([str(item) for item in data[var]])
+
+    else:
+        quit("Invalid propeller map format provided")
+
+    # store formatted data into NamedValues object
+    write_data = NamedValues()
+    for _, key in enumerate(data):
+        write_data.set_val(header_names[key], data[key], default_units[key])
```

Comment:
```suggestion
    for key in data:
        write_data.set_val(key.value, data[key], default_units[key])
```

Reply:
You are right. So, `header_names` is not needed.